django/db/models.py

```diff
@@ -775,15 +959,16 @@ def get_properties(obj):
     """
     Custom method to grab all of the fields _and_ properties on model instances
     """
-    
-    attrs     = {}
+    attrs = {}
     MethodWrapperType = type(object().__hash__)
 
     for slot in dir(obj):
         try:
             attr = getattr(obj, slot)
 
-            if ( slot.find('_') == 0 or slot == '__dict__' or slot == '__class__' 
+            if ( slot.find('_') == 0 
```

Comment:
Yes, I'll update this to clean up the logic.
This method was an attempt to make the get_properties operation django agnostic - newer versions of Django include an improved api for getting the fields (and properties).
aside - The motivation for this code was to translate the model instance into a dict, so that the dict could be associated with metadata. Tighter integration of the metadata into the models would make this process cleaner. However, this is all very ORM centric, whereas a separation of the ORM and the metadata into app layers would be even better, imho.